ui/app/actions.js

```diff
@@ -1194,6 +1243,12 @@ function cancelTx (txData) {
         dispatch(actions.clearSend())
         dispatch(actions.completedTx(txData.id))
         dispatch(actions.hideLoadingIndication())
+
+        if (global.METAMASK_UI_TYPE === ENVIRONMENT_TYPE_NOTIFICATION &&
+          !hasUnconfirmedTransactions(getState())) {
+          return global.platform.closeCurrentWindow()
```

Comment:
Where this inlined into a few actions now, would it make sense to move this into an action of its own?

```js
return dispatch(closeCurrentNotificationWindow())
```

Or maybe

```js
if (shouldCloseNotificationWindow()) {
    return dispatch(closeNotification window())
}
```

Reply:
See https://github.com/MetaMask/metamask-extension/issues/4866